components/supervisor/frontend/src/index.ts

```diff
@@ -4,47 +4,68 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-require('../public/index.css');
+require('../src/shared/index.css');
+require("reflect-metadata");
 
-import "reflect-metadata";
-import { createGitpodService, GitpodClient } from "@gitpod/gitpod-protocol";
-import { serverUrl, startUrl } from "./urls";
-import * as heartBeat from "./heart-beat";
-import * as LoadingFrame from "./loading-frame";
-import * as IdeFrame from "./ide-frame";
-import * as GitpodServiceClient from "./gitpod-service-client";
-import { SupervisorServiceClient } from "./supervisor-service-client";
-import { JsonRpcProxyFactory } from '@gitpod/gitpod-protocol/lib/messaging/proxy-factory';
+import { createGitpodService } from "@gitpod/gitpod-protocol";
+import * as GitpodServiceClient from "./ide/gitpod-service-client";
+import * as heartBeat from "./ide/heart-beat";
+import * as IDEService from "./ide/ide-service-impl";
+import * as LoadingFrame from "./shared/loading-frame";
+import { SupervisorServiceClient } from "./ide/supervisor-service-client";
+import { serverUrl, startUrl } from "./shared/urls";
+import * as IDEWebSocket from "./ide/ide-web-socket";
 
+window.gitpod = {
+    service: createGitpodService(serverUrl.toString())
+};
+IDEWebSocket.install();
```

Comment:
Just to reduce the probability that one of the transitive modules would try to use the WebSocket constructor on load.

Reply:
It is on the purporse, that the gitpod service is using the original websocket, which actually was broken because of https://github.com/gitpod-io/gitpod/pull/1963#discussion_r502765938